3_visualize/src/plot_gradient_legend.R

```diff
@@ -0,0 +1,98 @@
+plot_gradient_legend <- function(ensemble_data, site_info){
+  
+  # filter data to a focal time period
+  date_start <- as.Date("2021-06-28")
+  
+  threshold = 23.89 # C
+  
+  
+  legend_df <- ensemble_data %>%
+    filter(model_name == 'DA') %>%
+    mutate(obs_max_temp_f = c_to_f(obs_max_temp_c),
+           pred_max_temp_f = c_to_f(max_temp),
+           facet_title = "Legend"
+    ) %>%
+    # subset to time period of interest
+    filter(time %in% seq.Date(from = date_start, to = date_start, by = "1 day")) %>%
+    # and only Lordville
+    filter(seg_id_nat == "1573") %>%
+    # filter to 1 day out predictions only
+    filter(lead_time == 1) 
+  
+  # Calculate mean predicted temp for annotation 
+  mean_pred_t <- mean(legend_df$pred_max_temp_f, na.rm = T)
+  
+  # plot 1-day out predictions with mean prediction and observed
+  legend_df %>%
+    ggplot(
+      aes(
+        x = time,
+        y = pred_max_temp_f
+      )) +
+    labs(x = "",
+         y = "") +
+    # panel to label legend
+    facet_grid(~ facet_title) + 
+    # threshold line
+    geom_hline(yintercept = c_to_f(threshold),
+               linetype = "dashed",
+               color = "orangered",
+               size = .6,
+               alpha = 0.8) +
+    stat_gradientinterval(.width = c(0.1),
+                          shape = NA,
+                          width = 0.95,
+                          # sub values over threshold with NA
+                          aes(fill = ifelse(stat(y) > c_to_f(threshold), NA, stat(y))), 
+                          size = 0.5,
+                          p_limits = c(0.2, 0.3)) +
+    # gradient color scale, using red for NA (over threshold)
+    scico::scale_fill_scico(palette = "lapaz", 
+                            begin = 0.5,
+                            end = 0.7,
+                            na.value = "orangered") +
+    scico::scale_color_scico(palette = "lapaz", 
+                            end = 0.5,
+                            na.value = "orangered") +
+    # change alpha so that end of confidence interval shows and doesn't fade away
+    scale_slab_alpha_continuous(
+      range = c(.15, 1) #default: 0,1
```

Comment:
let's change this back to `range = c(0,1)`. we could either pull the 90%CI from the [old data](https://code.usgs.gov/wma/wp/forecast-preprint-code/-/blob/main/in/all_mods_with_obs.rds) to restrict what is displayed, add the black 90%CI bar using `.width`, or accept that it shows the full range of predicted values.

Reply:
added this comment to the issue, let's go ahead and change the range for now and leave it at that